src/main/resources/static/common/js/orders.js

```diff
@@ -51,146 +51,154 @@ app.controller('orderHistoryController', function($rootScope, $http, $filter) {
-				$http({
-					method: 'PUT',
-					url: '/proxy/v1/orders/'+history.order.orderNumber,
-					data: {
-						"status": history.order.status,
-						"reviewedByPRCH": true,
-						"reviewedByAPV": history.order.reviewedByAPV
-					},
-					headers: { 'Content-Type': 'application/json' }
-				}).then(function(result) {
-					angular.element(".new_label" + index).css("visibility", "hidden");
-					//Update icon
-					getUnreviewedAmount($http,$rootScope,$filter);
-				}).catch(function(result) {
-					console.log(result);
-				});
-			}
-
-			var introduceIncorrectNumberBug = false;
-			$.each($rootScope.deBugs,function(i,debug){
-	            if(debug.demoBugsType === "INCORRECT_NUMBER_OF_ITEMS_IN_SUMMARY_OF_PENDING_ORDER"){
-	            	introduceIncorrectNumberBug = true;
-	                return false;
-	            }
-	        });
+    history.showOrderDetail = {'show':false}
+
+    history.openOrderDetail = function(index, orderNumber){
+        $http({
+            method: 'GET',
+            url: '/proxy/v1/orders/'+orderNumber
+        }).then(function(result) {
+            history.order = result.data.data;
+
+            if(!history.order.reviewedByPRCH){
+                $http({
+                    method: 'PUT',
+                    url: '/proxy/v1/orders/'+history.order.orderNumber,
+                    data: {
+                        "status": history.order.status,
+                        "reviewedByPRCH": true,
+                        "reviewedByAPV": history.order.reviewedByAPV
+                    },
+                    headers: { 'Content-Type': 'application/json' }
+                }).then(function(result) {
+                    angular.element(".new_label" + index).css("visibility", "hidden");
+                    //Update icon
+                    getUnreviewedAmount($http,$rootScope,$filter);
+                }).catch(function(result) {
+                    console.log(result);
+                });
+            }
+
+            var introduceIncorrectNumberBug = false;
+            $.each($rootScope.deBugs,function(i,debug){
+                if(debug.demoBugsType === "INCORRECT_NUMBER_OF_ITEMS_IN_SUMMARY_OF_PENDING_ORDER"){
+                    introduceIncorrectNumberBug = true;
+                    return false;
+                }
+            });
 
             if(introduceIncorrectNumberBug && history.order.status === "PROCESSED"){
-            	history.totalItemQuantity = 0;
+                history.totalItemQuantity = 0;
             }else{
-            	var totalAmount = 0;
-            	$.each(history.order.orderItems,function(i,item){
+                var totalAmount = 0;
+                $.each(history.order.orderItems,function(i,item){
                     totalAmount += item.quantity;
                 });
 
-            	history.totalItemQuantity = totalAmount;
+                history.totalItemQuantity = totalAmount;
+            }
+
+            if(history.order.status === "CANCELED") {
```

Comment:
I used a stupid but reasonable method way to distinguish the comment type.
For the backend, the REST API should give user the detail in comments about which item is not found or the inventory of which item is not enough.
If there is any concren we can talk about it on scrum.

Reply:
The only solution I can think of is to add another field in the response message to indicate which error it is. But I think it might be too much for backend users. Maybe let's keep this for now since we are not likely to modify this message very often.